src/excmds.ts

```diff
@@ -2824,6 +2371,46 @@ export async function fillcmdline_tmp(ms: number, ...strarr: string[]) {
     )
 }
 
+/** @hidden **/
+const cmdframe_fns = {
+    "accept_line": ["process", []],
```

Comment:
I haven't had time to review this fully yet but this stuff looks pretty terrifying 👻. I'd like to avoid this kind of hard coded un-typed stuff if it's at all possible.

Reply:
I like the dot namespacing. We should perhaps focus on getting the user facing commands right so we don't have to change them later. The ugly code will be fixed for free I hope if V2 ever happens - most of this ugliness is because we don't have a generalised ex_commands_macros.py, right?

So if the dot style thing is compatible with the new way described in that branch that's hanging around I think we should plough ahead with what this PR currently has.

Reply:
>We should perhaps focus on getting the user facing commands right so we don't have to change them later.

I agree.

>most of this ugliness is because we don't have a generalised ex_commands_macros.py, right?

I'm not sure I understand what you mean. I think the main problem is wanting to use a special character to namespace things because it forces us to declare namespaced functions dynamically if we want to keep our dispatching code the way it is. I guess another solution that I hadn't envisaged is this:
- Finish & merge https://github.com/tridactyl/tridactyl/pull/1026 , this will enable access to type info for functions that aren't in excmds.ts from parsers/exmode.ts
- Move im./ex. functions to their own file
- Modify the excmd parser to use type info from metadata.ts instead of the one generated by the python macro processor.
- Have the [excmd parser](https://github.com/tridactyl/tridactyl/blob/f94fb9aa3dc74b19616313f0cfb97a251380a5a4/src/parsers/exmode.ts#L51) split `func` on dots. If there are no dots, use the regular excmd-dispatching code. If there are dots, use messageActiveTab to call the function in the right context (either content or command line).

Does that seem reasonable/more desirable than what we currently have?

Reply:
Just keep it the ugly way this PR was going to do before I complained and we can work out how to tidy it up later. I don't want to spend too much time thinking about some relatively contrived solution that will just be torn out when/if v2 happens.